api/applications.go

```diff
@@ -37,15 +39,103 @@ type ApplicationListResponse struct {
 	Icon        string `json:"icon"`
 }
 
+// DeleteApplications delete application
+func DeleteApplications(c *gin.Context) {
+	idParam := c.Param("id")
+	id, err := strconv.ParseUint(idParam, 10, 32)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Invalid id=%q", idParam),
+			Error:   err.Error(),
+		})
+		return
+	}
+	// Build application model TODO refactor
+	db := model.GetDB()
+	app := &model.Application{
+		ID: uint(id),
+	}
+	var deployments []*model.Deployment
+	db.Model(app).Related(&deployments, "Deployments")
+	log.Debugf("Associated deployments: %#v", deployments)
+	organization := auth.GetCurrentOrganization(c.Request)
+	err = db.Model(organization).Related(app).Error
+	if err != nil {
+		log.Errorf("Error getting application: %s", err.Error())
+		c.JSON(http.StatusBadRequest, components.ErrorResponse{
+			Code:    http.StatusBadRequest,
+			Message: "Error getting application",
+			Error:   err.Error(),
+		})
+		return
+	}
+	app.Deployments = deployments
```

Comment:
Why do we read deployments first before knowing if the application exists at all?

The flow rather should be, shouldn't it:

db.Model(organization).Related(app)

db.Model(app).Related(&app.Deployments, "Deployments")